users/vieting/experiments/switchboard/ctc/feat/experiments.py

```diff
@@ -776,6 +784,144 @@ def run_scf_baseline():
+        "final_epochs": 0,
+    }
+
+    perturbation_args = [
+        {"speed": {"prob": 0.6, "minimum": 0.8, "maximum": 1.2}},
+        {"speed": {"prob": 0.6, "minimum": 0.7, "maximum": 1.3}},
+        {"speed": {"prob": 0.5, "minimum": 0.9, "maximum": 1.1}},
+        {"speed": {"prob": 0.5, "minimum": 0.8, "maximum": 1.2}},
+        {"speed": {"prob": 0.5, "minimum": 0.7, "maximum": 1.3}},
+        {"speed": {"prob": 0.4, "minimum": 0.9, "maximum": 1.1}},
+        {"speed": {"prob": 0.4, "minimum": 0.8, "maximum": 1.2}},
+        {"speed": {"prob": 0.4, "minimum": 0.7, "maximum": 1.3}},
+        {"tempo": {"prob": 0.4, "minimum": 0.9, "maximum": 1.1}},
+        {"tempo": {"prob": 0.4, "minimum": 0.8, "maximum": 1.2}},
+        {"tempo": {"prob": 0.4, "minimum": 0.7, "maximum": 1.3}},
+        {"tempo": {"prob": 0.5, "minimum": 0.9, "maximum": 1.1}},
+        {"tempo": {"prob": 0.5, "minimum": 0.8, "maximum": 1.2}},
+        {"tempo": {"prob": 0.5, "minimum": 0.7, "maximum": 1.3}},
+        {"tempo": {"prob": 0.6, "minimum": 0.9, "maximum": 1.1}},
+        {"tempo": {"prob": 0.6, "minimum": 0.8, "maximum": 1.2}},
+        {"tempo": {"prob": 0.6, "minimum": 0.7, "maximum": 1.3}},
+        {"preemphasis": {"prob": 0.9, "minimum": 0.9, "maximum": 1.0}},
+        {"preemphasis": {"prob": 0.9, "minimum": 0.8, "maximum": 1.0}},
+        {"preemphasis": {"prob": 0.8, "minimum": 0.9, "maximum": 1.0}},
+        {"preemphasis": {"prob": 0.8, "minimum": 0.8, "maximum": 1.0}},
+        {"codecs": [{"encoding": "ULAW", "prob": 0.4}]},
+        {"codecs": [{"encoding": "ULAW", "prob": 0.6}]},
+        {"non_linearity": {"prob": 0.4, "minimum": 0.1, "maximum": 0.3}},
+        {"non_linearity": {"prob": 0.4, "minimum": 0.2, "maximum": 0.4}},
+        {"non_linearity": {"prob": 0.6, "minimum": 0.1, "maximum": 0.3}},
+        {"non_linearity": {"prob": 0.6, "minimum": 0.2, "maximum": 0.4}},
+    ]
+
+    def process_args(args):
+        """
+        Process the argument dictionary to generate a key string and a report string.
+
+        Args:
+            args (dict): The argument dictionary to process.
+
+        Returns:
+            tuple: A tuple containing the key string and the report string.
+        """
+        arg_exp_name = list(args.keys())
+        arg_values = list(args.values())
+        key_string = ""
+        report_values = ""
+
+        for i in range(len(arg_exp_name)):
+            key = arg_exp_name[i]
+            value = arg_values[i]
+
+            if key in ["speed", "tempo", "preemphasis", "non_linearity"]:
+                key_string += f"{key}_{value['prob']}_{value['minimum']}_{value['maximum']}_"
+                report_values += f"{key}: '{value['prob']}_{value['minimum']}_{value['maximum']}' "
+            elif key == "codecs":
+                key_string += f"{key}_{value[0]['encoding']}_{value[0]['prob']}_"
+                report_values += f"{key} (encoding: {value[0]['encoding']}): '{value[0]['prob']}' "
```

Comment:
Why only index 0? I'd argue if you have multiple codecs, you'd also want the alias to represent this.